packages/plugins/plugin-project/src/components/ProjectSettings.tsx

```diff
@@ -3,135 +3,177 @@
+        return;
+      }
+
+      view.query.grammar = newQueryString;
+      const newQuery = evalQuery(newQueryString);
+      const newSchema = await resolveSchemaWithClientAndSpace(client, space, newQuery.ast);
+      if (!newSchema) {
+        return;
+      }
+
+      const newView = createView({
+        query: newQueryString,
+        jsonSchema: Type.toJsonSchema(newSchema),
+        presentation: Obj.make(Type.Expando, {}),
+      });
+      view.projection = Obj.getSnapshot(newView).projection;
+      setSchema(() => newSchema);
+    },
+    [view, schema],
+  );
 
-const getLabel = (project: DataType.Project) => {
-  const { mailboxQuery } = parseProjectPreset(project) ?? {};
-  if (!mailboxQuery) {
-    return '';
-  }
+  const handleToggleField = useCallback((view: DataType.View) => {
+    setExpandedId((prevExpandedId) => (prevExpandedId === view.id ? undefined : view.id));
+  }, []);
 
-  let label = '';
-  QueryAST.visit(mailboxQuery, (node) => {
-    if (node.type !== 'select') {
-      return;
-    }
-    if (node.filter.type !== 'object') {
-      return;
-    }
-    if (node.filter.props.properties.type !== 'object') {
-      return;
-    }
-    if (node.filter.props.properties.props.labels.type !== 'contains') {
-      return;
-    }
+  const handleDelete = useCallback(
+    (view: DataType.View) => {
+      if (view.id === expandedId) {
+        setExpandedId(undefined);
+      }
 
-    label = node.filter.props.properties.props.labels.value;
-  });
+      const index = project.collections.findIndex((ref) => ref.target === view);
+      project.collections.splice(index, 1);
+      space?.db.remove(view);
+    },
+    [expandedId, project.collections, space],
+  );
 
-  return label;
+  const handleAdd = useCallback(() => {
+    const view = createView({
+      query: Query.select(Filter.type(DataType.Task)),
+      jsonSchema: Type.toJsonSchema(DataType.Task),
+      presentation: Obj.make(DataType.Collection, { objects: [] }),
+    });
+    project.collections.push(Ref.make(view));
+    setExpandedId(view.id);
+  }, [project]);
+
+  return (
+    <div role='none' className={mx('plb-cardSpacingBlock overflow-y-auto', classNames)}>
+      <h2 className={mx(inputTextLabel, cardText)}>{t('views label')}</h2>
+
+      <List.Root<DataType.View>
+        items={views}
+        isItem={Schema.is(DataType.View)}
+        getId={(view) => view.id}
+        onMove={handleMove}
+      >
+        {({ items: views }) => (
+          <>
+            <div role='list' className={mx(listGrid, cardChrome)}>
+              {views.map((view) => (
+                <List.Item<DataType.View>
+                  key={view.id}
+                  item={view}
+                  classNames={listItemGrid}
+                  aria-expanded={expandedId === view.id}
+                >
+                  <div role='none' className={mx(subtleHover, listItemGrid, 'rounded-sm cursor-pointer min-bs-10')}>
```

Comment:
nit: i try to order classes from most significant to least (i.e., min-bs-10) first, although this seems like an overly specific class? (10?)

Reply:
I copied this list styling from ViewEditor, it probably deserves a cleanup